server/integration/infinispan/src/main/resources/schema/jboss-infinispan-core_9_2.xsd

```diff
@@ -371,6 +371,115 @@
         </xs:sequence>
     </xs:complexType>
 
+	<xs:complexType name="counters">
+		<xs:sequence>
+			<xs:element name="strong-counter" type="tns:strong-counter"
+				minOccurs="0" maxOccurs="1">
+				<xs:annotation>
+					<xs:documentation>
+						Defines strong counter
+					</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="weak-counter" type="tns:weak-counter"
+				minOccurs="0" maxOccurs="1">
+				<xs:annotation>
+					<xs:documentation>
+						Defines strong counter
+					</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+		</xs:sequence>
+		<xs:attribute name="num-owners" type="xs:int" default="2"
```

Comment:
For the default values you should placeholders like other defaults in the file. For example `${Server.CacheContainer.start}`. To find the placeholder for the counter defaults, look in `target/defaults.properties` after making the server and you should see the placeholder text and value for `num-owners` and all other configured resources that have default values defined

Reply:
Although, looking at the `num-owners` resource, it seems you haven't set the default value via `setDefaultValue(...)`

Reply:
@ryanemerson I put the placeholders in schema file, build the server, but then I do not see the values generated in target/default.properties?

Reply:
How can this work, how can processing build program figure out the defaults for the placeholders?

Reply:
Do I have to name placeholders according to some naming scheme?

Reply:
This is all done by the [defaults plugin](https://github.com/infinispan/infinispan/tree/master/plugins/infinispan-defaults-maven-plugin). During the build process, this plugin extracts the defined defaults of all infinispan `AttributeDefinition` or jboss `SimpleAttributeDefinition` and outputs them to the `target/defaults.properties` file. 

Property keys are the name of the class which contains the definition (minus Resource at the end), concatenated with the name of the attribute. Furthermore in the case of jboss resources, the key is pre-pended with `Server`. So if you update the NUM_OWNERS to the following:

```java
static final SimpleAttributeDefinition NUM_OWNERS = new SimpleAttributeDefinitionBuilder(ModelKeys.NUM_OWNERS,
            ModelType.LONG, false)
            .setXmlName(Attribute.NUM_OWNERS.getLocalName())
            .setDefaultValue(new ModelNode(2)) // You need to add this for a default value to be defined
            .setAllowExpression(false)
            .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
            .build();
```

Then the placeholder key, which you should use in the xsd will be `${Server.CacheContainerCounters.num_owners}`. 

FYI all of the successfully extracted defaults for the server will be located at `$INFINISPAN_ROOT/server/integration/infinispan/target/defaults.properties`

Reply:
Wow, awesome! Thanks for the explanation @ryanemerson